Adds support for `--ignore-errors` to qvm-clone CLI

conorsch opened the pull request:
The underlying API already supported the `ignore_errors` param in
Python, now we just expose that option to the `qvm-clone` CLI.

This PR ostensibly closes https://github.com/QubesOS/qubes-issues/issues/4809.

I realize there are no tests provided here. Happy to add some, although I might request a bit of guidance. :slightly_smiling_face:

conorsch committed:
Adds support for `--ignore-errors` to qvm-clone CLI

The underlying API already supported the `ignore_errors` param in
Python, now we just expose that option to the `qvm-clone` CLI.

marmarek commented:
Lack of tests is one issue - actually, [already existing tests for qvm-clone](https://github.com/QubesOS/qubes-core-admin-client/blob/master/qubesadmin/tests/tools/qvm_clone.py) tool now fails because of extra argument. It should be enough to copy `test_000_simple` and pass `--ignore-errors` argument there.

The other issue is lack of documentation (which also cause test build to fail):
```
Sphinx error:
Undocumented arguments for command 'qvm-clone': '--ignore-errors'
```
Pylint also complains about too long lines:
```
************* Module qubesadmin.tools.qvm_clone
qubesadmin/tools/qvm_clone.py:42:0: C0301: Line too long (88/80) (line-too-long)
qubesadmin/tools/qvm_clone.py:79:0: C0301: Line too long (114/80) (line-too-long)
```
Other than that, looks good, thanks!

conorsch committed:
Adds docs for clone ignore_errors, feeds linter

Addresses review comments, specifically:

  * updates qvm-clone tests to handle ignore_errors option
  * adds manpage reference to new --ignore-errors option
  * trims line length in qvm-clone changes

All tests should now be passing, including the new qvm-clone
functionality.

conorsch commented:
Thanks for the pointers, the test suite is robust and easy to work with. The qvm-clone tests are now passing, but I'm still [seeing errors](https://travis-ci.com/QubesOS/qubes-core-admin-client/jobs/193415100), which appear to be unrelated to this diff:

```
FAIL: test_000_write_backup_profile (tools.qvm_backup.TC_00_qvm_backup)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/QubesOS/qubes-core-admin-client/build/lib/qubesadmin/tests/tools/qvm_backup.py", line 39, in test_000_write_backup_profile
    self.assertEqual(profile.getvalue(), expected_profile)
AssertionError: 'compression: true\ndestination_path: /var[39 chars]ll\n' != '{compression: true, destination_path: /va[41 chars]l}\n'
+ {compression: true, destination_path: /var/tmp, destination_vm: dom0, include: null}
- compression: true
- destination_path: /var/tmp
- destination_vm: dom0
- include: null
```

For what it's worth, I'm seeing those same errors on master (2f1be166359b4c516c1dd60ccdb7f481274d189e) locally, but that may be a problem with my env. I'm running Python 3.5.3. Happy to work on resolving those diff comparisons, either here or in a separate PR.

marmarek commented:
> The qvm-clone tests are now passing, but I'm still [seeing errors](https://travis-ci.com/QubesOS/qubes-core-admin-client/jobs/193415100), which appear to be unrelated to this dif

Yes, seems unrelated.
